### DISCUSSION TITLE
when i send a request report 400

### DISCUSSION THREAD

**Discussion by poohlaha** [hyperium/hyper#3248] [Q&A]
I use demo code(https://github.com/hyperium/hyper/blob/master/examples/web_api.rs)
```
use bytes::{Bytes};
use tokio::runtime::Runtime;
use hyper::{header, Request, Method};
use http_body_util::{Full};
use tokio::net::{TcpStream};

static URL: &str = "http://jsonplaceholder.typicode.com/users";

async fn client_request_response()  {
    let req = Request::builder()
        .method(Method::GET)
        .uri(URL)
        .header(header::CONTENT_TYPE, "application/json")
        .body(Full::new(Bytes::new()))
        .unwrap();

    let url: hyper::Uri = URL.parse().unwrap();
    let host = url.host().expect("uri has no host");
    let port = url.port_u16().unwrap_or(80);
    let stream = TcpStream::connect(format!("{}:{}", host, port)).await.unwrap();

    let (mut sender, conn) = hyper::client::conn::http1::handshake(stream).await.unwrap();

    tokio::task::spawn(async move {
        if let Err(err) = conn.await {
            println!("Connection error: {:?}", err);
        }
    });

    let web_res = sender.send_request(req).await.unwrap();
    println!("{:?}", web_res)
}

fn main() {
    let runtime = Runtime::new().unwrap();
    runtime.block_on(async {
        client_request_response().await;
    });
}
```

when i run get response:
Response { status: 400, version: HTTP/1.1, headers: {"server": "cloudflare", "date": "Tue, 13 Jun 2023 07:15:13 GMT", "content-type": "text/html", "content-length": "155", "connection": "close", "cf-ray": "-"}, body: Body(Streaming) }

then the `http://jsonplaceholder.typicode.com/users` results:
![image](https://github.com/hyperium/hyper/assets/10954901/6ed08337-6ce8-4d32-a388-198e52a34382)

and i want to send a https request, use `hyper-tls`, but not find `Client`, what should I do? Or is there a good solution? The documentation is not detailed at all, and there is no manual in README.md, which is really time-consuming.

**Comment by seanmonstar** [comment]
Are you specifically interested in the lower level details that hyper provides? Or do you just want a fuller-featured HTTP client? If the latter, I recommend trying https://crates.io/crates/reqwest.
